src/ssl.c

```diff
@@ -30211,72 +30268,43 @@ WOLFSSL_RSA* wolfSSL_RSA_generate_key(int len, unsigned long e,
         WOLFSSL_MSG("memory error");
     }
     else {
-        if (wolfSSL_RSA_generate_key_ex(rsa, len, bn, NULL) != SSL_SUCCESS){
-            wolfSSL_RSA_free(rsa);
-            rsa = NULL;
+        for (;;) {
+            int gen_ret = wolfSSL_RSA_generate_key_native(rsa, len, bn, NULL);
+            if (gen_ret == WOLFSSL_ERROR_NONE)
+                break;
+#ifdef HAVE_FIPS
```

Comment:
The more rigorous check for possible primality was added for FIPSv2. This should be

    #if defined(HAVE_FIPS_VERSION) && (HAVE_FIPS_VERSION >= 2)

Reply:
The lower level function this calls for the key generation is switching on HAVE_FIPS. this is fine. If they are using FIPSv1, that isn't going to return the random error anyway.